weather.station.server/weather.station.server/Controllers/HomeController.cs

```diff
@@ -1,25 +1,47 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
 using weather.station.server.Data;
+using weather.station.server.Models;
+using weather.station.server.Models.ViewModels;
 
 namespace weather.station.server.Controllers
 {
     public class HomeController : Controller
     {
-        private WeatherStationServerContext _context;
+        private readonly WeatherStationServerContext _context;
 
         public HomeController(WeatherStationServerContext context)
         {
             _context = context;
         }
 
         [HttpGet]
-        public IActionResult Index()
+        public IActionResult Index(DateTime? sliderTime)
         {
-            return View();
+            var time = sliderTime ?? DateTime.Now; //if date time is null, datetime is now
+
+            ICollection<WeatherUpdate> weatherUpdates = new List<WeatherUpdate>();
+
+            if (_context.WeatherUpdate.Any())
+            {
+                var updates = _context.WeatherUpdate
+                    .Include(d => d.Device)
+                    .GroupBy(i => i.DeviceId);
+
+                weatherUpdates = updates
+                    .Select(g => g.OrderByDescending(d => d.TimeStamp).First(d => d.TimeStamp < time))
```

Comment:
Dit ziet er een beetje uit als een rare query, kan iemand dit toelichten?

EDIT: Nvm, heb een haakje gemist in  mn hoofd.